deprecation.py

```diff
@@ -165,6 +172,11 @@ def deprecated(deprecated_in=None, removed_in=None, current_version=None,
         elif (deprecated_in
               and current_version >= version.parse(deprecated_in)):
             is_deprecated = True
+    elif isinstance(removed_in, date):
```

Comment:
If you move this branch to be the first one,you don't need the `and not isinstance(...)` on the version conditional.

Reply:
In those cases where `removed_in` is specified but `current_version` is not, we lose the documented behavior that is `current_version` is not specified, we always raise `DeprecationError`.

Reply:
Got it